src/main/java/uk/gov/pay/ledger/transaction/service/TransactionService.java

```diff
@@ -88,4 +101,50 @@ public void upsertTransactionFor(EventDigest eventDigest) {
         TransactionEntity transaction = transactionEntityFactory.create(eventDigest);
         transactionDao.upsert(transaction);
     }
+
+    public TransactionEventResponse findTransactionEvents(String externalId, String gatewayAccountId) {
+        Map<String, TransactionEntity> transactionEntityMap = getTransactionsAsMap(externalId, gatewayAccountId);
+
+        if (transactionEntityMap.isEmpty()) {
+            throw new BadRequestException(format("Transaction with id [%s] not found", externalId));
```

Comment:
should the gatewayAccountId be logged as well given both are used to query db?

Reply:
Publicapi usually relays same message to services. Don't want to expose gateway_account_id in the response

Reply:
Fair enough - I thought about logging tbh.